salt/modules/solarispkg.py

```diff
@@ -360,7 +360,10 @@ def install(name=None, sources=None, saltenv="base", **kwargs):
         cmd_prefix = ["/usr/sbin/pkgadd", "-n", "-a", adminfile]
 
         # Only makes sense in a global zone but works fine in non-globals.
-        if kwargs.get("current_zone_only") == "True":
+        if (
+            kwargs.get("current_zone_only") is True
+            or kwargs.get("current_zone_only") == "True"
+        ):
```

Comment:
```suggestion
        if kwargs.get("current_zone_only") in (True, "True"):
```

Reply:
Or even:
```suggestion
        if str(kwargs.get("current_zone_only")) == "True"):
```

Reply:
I'd say the first suggestion would be the more correct

Reply:
Did the first suggestion